blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/AbstractPipelineImpl.java

```diff
@@ -210,7 +209,58 @@ public static String getRecursivePathFromFullName(BluePipeline pipeline){
 
     @Override
     public Container<Resource> getActivities() {
-        return Containers.fromResource(getLink(), Lists.newArrayList(Iterators.concat(getQueue().iterator(), getRuns().iterator())));
+
+        return new Container<Resource>(){
+            @Override
+            public Iterator<Resource> iterator() {
+                return Lists.newArrayList(Iterators.concat(getQueue().iterator(), getRuns().iterator())).iterator();
```

Comment:
@vivek won't creating a new `ArrayList` like this read all the items from the iterator and add them to the list? `Iterators.concat()` would be more appropriate?